### PR TITLE
Set quantity values with units

### LINKED ISSUE #629: Allow setting parameter/molecule/quantity values with units
Currenty: `setQuantityValues <- function(quantities, values)`

Proposal: `setQuantityValues <- function(quantities, values, units = NULL)`

and with `units = NULL`, base units are used.

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #629

### CODE REVIEW

File: R/utilities-molecule.R [hunk 1]
```diff
@@ -81,11 +82,38 @@ getMolecule <- function(path, container, stopIfNotFound = TRUE) {
...
+#'
+#' @examples
+#'
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' setMoleculeValuesByPath("Organism|VenousBlood|Plasma|CYP3A4", 1, sim)
+#'
+#' setMoleculeValuesByPath(c("Organism|VenousBlood|Plasma|CYP3A4", "Organism|ArterialBlood|Plasma|CYP3A4"), c(2, 3), sim, units = c("µmol", "mmol"))
+#' @export
+setMoleculeValuesByPath <- function(moleculePaths, values, simulation, stopIfNotFound = TRUE, units) {
```
Comment (PavelBal) [inline R/utilities-molecule.R:109]: We had this for parameters but not for molecules.

File: R/utilities-units.R [hunk 2]
```diff
@@ -280,9 +280,10 @@ getUnitsEnum <- function() {
     x <- getUnitsForDimension(dimension = dimension)
     return(enum(replace(x, x == "", "Unitless")))
   })
-  names(units) <- sapply(dimensions,function(str){
-    str <- gsub(pattern = "[(]",replacement = "[",x = str)
-    str <- gsub(pattern = "[)]",replacement = "]",x = str)})
+  names(units) <- sapply(dimensions, function(str) {
```
Comment (PavelBal) [inline R/utilities-units.R:283]: Just styler

File: R/utilities-quantity.R [hunk 3]
```diff
@@ -113,19 +125,31 @@ setQuantityValues <- function(quantities, values) {
 #'
 #' setParameterValuesByPath(list("Organism|Liver|Volume", "Organism|Liver|A"), c(2, 3), sim)
 #' @export
-setQuantityValuesByPath <- function(quantityPaths, values, simulation, stopIfNotFound = TRUE) {
+setQuantityValuesByPath <- function(quantityPaths, values, simulation, stopIfNotFound = TRUE, units = NULL) {
```
Comment (msevestre) [inline R/utilities-quantity.R:128]: looking good. stopInfNotFound is in general OUR LAST parameter. Here we are changing the signatures `logic` .
shouldn't we put `units` before `stopIfNotFound`  If people have used this flag, one can argue that they would have named it explicitly to set it to FALSE

**Comment by msevestre** [comment]
@PavelBal I have fixed the code to PR is working now. Let me know your thoughts on the order of parameters.

File: R/utilities-quantity.R (hunk 3 above)
Comment (PavelBal) [reply R/utilities-quantity.R:128]: I couldn't decide what is better - keep the logic by putting `stopIfNotFound` at the end, or maintain compatibility with code that uses the function without specifying argument names. But I think the method has not been used extensively so far, so let's keep the logic. I will update.

**Comment by PavelBal** [comment]
@msevestre `stopIfNotFound` is now always the last parameter. Ready to merge unless you have other comments.

**Pull request merged**